### DISCUSSION TITLE
spm "ready for prime time"

### DISCUSSION THREAD

**Discussion by Remi-Gau** [spm/spm#28] [General]
I am starting to encounter errors in CI when trying to use SPM 12 with macOS latest (Sonoma): 

Operating System
-   macOS
-   14.4.1
-   23E224

https://github.com/cpp-lln-lab/CPP_ROI/actions/runs/9004122591/job/24736368707#step:7:34
```
   {Error using spm_slice_vol
  spm_slice_vol.c not compiled - see Makefile
```

Trying to compile SPM does not seem to help:
https://github.com/cpp-lln-lab/CPP_ROI/actions/runs/9004450583/job/24737447777?pr=127#step:4:3720

This made me think that I should probably start using the spm repo instead of the spm12 one but I am not sure if the spm is ready to be used "in production" or if it should still be considered "bleeding edge" at the moment.

If the latter, would you have any pointers to help fix those macos-latest related issues.

Let me know if you want me to provide an minimal workflow to reproduce the error.

**Comment by tierneytim** [comment]
Great discussion point!

If you check out our actions, you will see we are also struggling to get our testing working well on newer macOS. It's a pain but our transition to a fully automated test system has been tricky. 

But to clarify is this only a problem you face in CI or is SPM12 struggling on your local machine as well? 

On the broader point of whether you should you leave spm12 behind and move to spm, I think you should. We will ultimately struggle to provide support for any issues that are specific to what is now a very old release of SPM.

**Reply by Remi-Gau** [reply]
I had not seen the workflow you have on the spm repo. Will check it out.

The problem is just in CI and only for macOS: most other things are still running smoothly with other OS and with Octave (only testing on linux though).

About leaving spm12 and moving to spm: are there going to be tagged releases we can pin in the downstream toolbox that use spm?

I think that it will be easier to ensure reproducibility for most maintainers than relying just on a commit shasum or a git submodule (I love the latter but I know most people don't share my sentiment).

**Comment by tierneytim** [comment]
> are there going to be tagged releases we can pin in the downstream toolbox that use spm?

Yup, we're still working out the exact schedule but up to  two times a year, with a dated (tagged) releases. First official one will be this year.

**Reply by Remi-Gau** [reply]
OK good to know.

Wills start working on "pre-release" workflows to set up to test on the default branch of the spm repo.
